sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala

```diff
@@ -530,3 +532,283 @@ private[columnar] case object LongDelta extends CompressionScheme {
     }
   }
 }
+
+/**
+ * Writes integral-type values with delta encoding and binary packing.
+ * The format is as follows:
```

Comment:
I just used the parquet spec./impl. as a reference; some parts of the impl. are modified to easily compute compression size in `gatherCompressibilityStats`.
https://github.com/Parquet/parquet-mr/blob/master/parquet-column/src/main/java/parquet/column/values/delta/DeltaBinaryPackingValuesWriter.java#L39